src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ExtractToComponentBehindCodeActionProvider.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+using Microsoft.AspNetCore.Razor.Threading;
+using Microsoft.CodeAnalysis.Razor.Logging;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Razor;
+internal sealed class ExtractToComponentBehindCodeActionProvider : IRazorCodeActionProvider
+{
+    private readonly ILogger _logger;
+
+    public ExtractToComponentBehindCodeActionProvider(ILoggerFactory loggerFactory)
+    {
+        if (loggerFactory is null)
+        {
+            throw new ArgumentNullException(nameof(loggerFactory));
+        }
```

Comment:
```suggestion

```

@davidwengier we're moving away from these checks on internal code right?

Reply:
Yep, as long as nullability analysis is enabled, we shouldnt need to null check. There are a few spots in the compiler that still aren't (like with the syntax tree below this) but for contructor parameters, definitely don't need them.

Reply:
Also, if we *do* need `ArgumentNullException` checks, there's a one-liner for that in the Razor code base:

```C#
ArgHelper.ThrowIfNull(loggerFactory);
```